plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -82,6 +85,36 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Adds the product template ID to the product if it doesn't exist.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WC_Product       $product The product.
+	 */
+	public function possibly_add_template_id( $response, $product ) {
+		if ( ! $product ) {
+			return $response;
+		}
+		if ( ! $product->meta_exists( '_product_template_id' ) ) {
+			/**
+			 * Allows to determine a product template id based on the product data.
+			 *
+			 * @since 9.1.0
+			 */
+			$product_template_id = apply_filters( 'woocommerce_product_editor_determine_product_template', '', $product );
```

Comment:
I think the filter name could be `woocommerce_product_editor_product_tempate_for_product`

Reply:
I think you missed something in your suggestion. Can you double-check?